lessons/git/git-collaboration-2in1/index.md

```diff
@@ -0,0 +1,291 @@
+      (use "git push" to publish your local commits)
+    nothing to commit, working tree clean
+
+Pro lepší pochopení, co dělají jednotlivé příkazy, a v jakém stavu můžou být soubory/změny, přikládáme tento diagram: 
+
+![Git workflow]({{ static('diagram.png') }})
+
+#### Log (`git log`)
+Teď, když za sebou máme první revizi, podívejme se, jak vypadá historie repozitáře, do kterého chceme přispívat.
+K tomu můžeme použít příkaz `git log`, jako první uvidíme naši poslední změnu a další jsou změny od jiných autorů,
+na které navazujeme.
+
+    $ git log
+    commit 0bdfbb2a2398fea179395a8dd303e9f672ef4dca
+    Author: Magdalena Dobromila <magdalena@rettigova.cz>
+    Date:   Tue Mar 28 19:27:08 2017 +0200
+
+        Přidáno mé jméno a povolání
+
+    commit 0f305972803131cb6c8637359fee8ede3005bba6
+    Merge: effa89b 175f4cc
+    Author: Oskar Hollmann <oskar@hollmann.me>
+    Date:   Tue Nov 8 20:17:00 2016 +0100
+
+        Merge pull request #1 from muzikovam/master
+        
+        Tady mas jmeno
+
+Příkaz `git log` má mnoho přepínačů, přehlednější historii změn můžeme získat pomocí `git log --oneline --graph --decorate`.
+
+### Posílání změn (`git push`)
+
+Nyní stačí požádat autora projektu, aby naše změny začlenil do repozitáře na GitHubu 
+a všichni uživatelé projektu, na kterém pracujeme, budou mít užitek z našich změn.
+Nemáme ale do původního repozitáře práva na zápis, musíme tedy poslat změny nejprve do naší kopie repozitáře,
+vizte obrázek na začátku.
+
+Udělejte si účet na GitHubu (jestli ho ještě nemáte), a pak jděte na adresu, kterou jste použili pro `git clone`. 
+Vlevo nahoře je tlačítko „Fork”, klikněte na něj. Tím se na GitHubu vytvoří vaše kopie repozitáře: 
+adresa by měla být něco jako `https://github.com/tvojejmeno/prezencka`.
+
+A teď, jak nahrát změny z našeho počítače na GitHub? Git si u každého repozitáře na našem počítači pamatuje adresy, 
+odkud se dají stahovat a kam se dají posílat změny. 
+Seznam těchto adres nám ukáže příkaz `git remote -v`. Třeba:
+
+    $ git remote -v
+    origin  https://github.com/asgeirrr/prezencka (fetch)
+    origin  https://github.com/asgeirrr/prezencka (push)
+
+Tenhle výstup znamená, že pod zkratkou *origin* se schovává adresa, ze kterého jsme repozitář naklonovali.
+
+Přidejme si podobnou zkratku pro vlastní repozitář na GitHubu. Můžeme ho pojmenovat např. *public*,
+aby nám bylo jasné, že do tohoto repozitáře můžeme nahrávat změny.
+
+    $ git remote add public https://github.com/tvojejmeno/prezencka 
+
+Zkontrolujme, že se nám to povedlo:
+
+    $ git remote -v
+    origin  git@github.com:asgeirrr/prezencka.git (fetch)
+    origin  git@github.com:asgeirrr/prezencka.git (push)
+    public  https://github.com/tvojejmeno/prezencka (fetch)
+    public  https://github.com/tvojejmeno/prezencka (push)
+
+Tolik k nastavení -- `git remote add` stačí udělat jednou pro každý repozitář. Pak už můžeme změny nahrávat pomocí: 
+
+    $ git push public master
```

Comment:
Ale nahoře se to s přidává https, to se divím. Snad to bude fungovat. Možná to .git na konci. Přidám to.